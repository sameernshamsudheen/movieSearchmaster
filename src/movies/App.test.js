import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';



afterEach(cleanup);

describe('Test cases for movies App', () => {

  it('should render the App and match the snapshot', () => {
    const component = renderer.create(
      <App />
    );
    let app = component.toJSON();
    const { getByText } = render(<App />);
    expect(screen.getByText("Movie Search")).toBeInTheDocument();
    expect(app).toMatchSnapshot();
  })


  it('should render input field', () => {
    const { component } = render(<App />);
    const input = screen.getByTestId("search-input");
    fireEvent.change(input, { target: { value: "action" } });
    expect(input.value).toBe("action");
  });

  it("shows no match found message when no genre is found", () => {
    const { getByPlaceholderText, getByText } = render(<App />);
    const input = screen.getByTestId("search-input");
    fireEvent.change(input, { target: { value: "not-existing-genre" } });
    expect(screen.getByText("No Match Found")).toBeInTheDocument();
  });

  it("shows the correct movie with the selected genre", () => {
    const { getByPlaceholderText, getByText } = render(<App />);
    const input = screen.getByTestId("search-input");
    fireEvent.change(input, { target: { value: "action" } });
    const movieList = screen.getByTestId("movie-list");
    expect(movieList).toHaveTextContent("1Inception8.8⭐");  });

  it("shows the top 3 movies with the selected genre", () => {
    const { getByPlaceholderText, getByText } = render(<App />);
    const input = screen.getByTestId("search-input");
    fireEvent.change(input, { target: { value: "drama" } });
    const movieList = screen.getByTestId("movie-list");
    expect(movieList).toHaveTextContent("1Remember the Titans7.6⭐2Godfather9.2⭐3Citizen Kane8.5⭐");
  });

  it("shows the selected genre name", () => {
    const { getByPlaceholderText, getByText } = render(<App />);
    const input = screen.getByTestId("search-input");
    fireEvent.change(input, { target: { value: "action" } });
    expect(screen.getByText("Genre : Action")).toBeInTheDocument();
  });

})


