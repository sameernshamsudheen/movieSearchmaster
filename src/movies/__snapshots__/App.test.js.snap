// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for movies App should render the App and match the snapshot 1`] = `
<div
  className="App"
>
  <h2>
    Movie Search
  </h2>
  <input
    data-testid="search-input"
    onChange={[Function]}
  />
  <div
    className="container"
  >
    <div>
      <h4>
        Enter Genre to Search
      </h4>
    </div>
  </div>
</div>
`;
